bundles/org.openhab.binding.myenergi/src/main/java/org/openhab/binding/myenergi/internal/handler/MyenergiBridgeHandler.java

```diff
@@ -0,0 +1,195 @@
+import org.openhab.binding.myenergi.internal.exception.AuthenticationException;
+import org.openhab.binding.myenergi.internal.exception.RecordNotFoundException;
+import org.openhab.binding.myenergi.internal.model.CommandStatus;
+import org.openhab.binding.myenergi.internal.model.EddiSummary;
+import org.openhab.binding.myenergi.internal.model.HarviSummary;
+import org.openhab.binding.myenergi.internal.model.MyenergiData;
+import org.openhab.binding.myenergi.internal.model.ZappiChargingMode;
+import org.openhab.binding.myenergi.internal.model.ZappiSummary;
+import org.openhab.core.config.core.status.ConfigStatusCallback;
+import org.openhab.core.config.core.status.ConfigStatusSource;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MyenergiBridgeHandler} is responsible for handling commands, which
+ * are sent to one of the channels.
+ *
+ * @author Rene Scherer - Initial contribution
+ */
+@NonNullByDefault
+public class MyenergiBridgeHandler extends BaseBridgeHandler implements ConfigStatusCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(MyenergiBridgeHandler.class);
+
+    private final MyenergiApiClient apiClient;
+
+    private @Nullable ScheduledFuture<?> devicePollingJob = null;
+
+    public MyenergiBridgeHandler(Bridge thing, HttpClientFactory httpClientFactory) {
+        super(thing);
+
+        // create a new httpClient, so that we can add our own digest authentication
+        apiClient = new MyenergiApiClient(httpClientFactory.createHttpClient("MyenergiBridge"));
+    }
+
+    public ThingUID getUID() {
+        return thing.getUID();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MyEnergiBridgeHandler");
+        MyenergiBridgeConfiguration config = getConfigAs(MyenergiBridgeConfiguration.class);
+
+        if (config.hubSerialNumber.isEmpty() || config.password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-missing-hubSerialNumber-or-password");
+            return;
+        }
+        if (config.refreshInterval < 1) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error-invalid-refresh-intervals");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            logger.debug("Login to MyEnergi API with hubSerialNumber: {}", config.hubSerialNumber);
```

Comment:
This type of logging should be part of the apiclient.  (seperation of concern) 

This class does not care if it needs to login, jump trough a ring of fire or anything else, it just calls the class to initialize.